How to create mac binary

andreievg opened the pull request:
Result can be found in: https://drive.google.com/drive/u/1/folders/1kmP-GLq-48sKPqpbW4nS_t6ethkQlixs

andreievg commented on server/README.md:
```diff
@@ -249,3 +249,34 @@ cargo run --bin remote_server_cli -- initialise-from-central -u 'user1:password1
+#!/bin/bash
+cd "$(dirname "$0")"
+chmod +x bin/remote_server
+xattr -cr bin/remote_server
```
Mac automatically adds certain attributes to executables that make them look like they are corrupt (for security), this removes it

andreievg commented on server/README.md:
```diff
@@ -249,3 +249,34 @@ cargo run --bin remote_server_cli -- initialise-from-central -u 'user1:password1
+# Create script to start server
+echo '
+#!/bin/bash
+cd "$(dirname "$0")"
```
Can double click on the .sh too run it from finder

andreievg commented on server/README.md:
````diff
@@ -249,3 +249,34 @@ cargo run --bin remote_server_cli -- initialise-from-central -u 'user1:password1
+cp omsupply-database m1_mac_sqlite/
+```
+
+In finder, right click on open_msupply_server, get info and make it run in terminal then `chmod +x m1_mac_sqlite/open_msupply_server.sh`
````
It's quite important to do this in the order, otherwise after unzipping downloaded .sh will have the attributes and won't be runnable

mark-prins replied:
wasn't sure what you mean by 'make it run in terminal'. Do you mean changing 'open with' to be terminal? Perhaps give some instructions if so?

mark-prins commented on server/README.md:
```diff
@@ -249,3 +249,34 @@ cargo run --bin remote_server_cli -- initialise-from-central -u 'user1:password1
+xattr -cr bin/remote_server
+APP_SERVER__DANGER_ALLOW_HTTP=true ./bin/remote_server' > m1_mac_sqlite/open_msupply_server.sh
+
+# Create database, make sure you have data file running locally (with correct credentials)
```
```suggestion
# Create sqlite database
# Before running this command, make sure you have data file running locally and update the port, site ID, and sync credentials
```

mark-prins commented on server/README.md:
```diff
@@ -249,3 +249,34 @@ cargo run --bin remote_server_cli -- initialise-from-central -u 'user1:password1
+# Create database, make sure you have data file running locally (with correct credentials)
+APP_SYNC__URL="http://localhost:2048" APP_SYNC__USERNAME="demo" APP_SYNC__PASSWORD_SHA256="d74ff0ee8da3b9806b18c877dbf29bbde50b5bd8e4dad7a3a725000feb82e8f1" APP_SYNC__SITE_ID=2  APP_SYNC__INTERVAL_SEC=10000000 APP_SYNC__CENTRAL_SERVER_SITE_ID=1 cargo run --bin remote_server_cli -- initialise-from-central -u 'Admin:pass'
+
+cp omsupply-database m1_mac_sqlite/
```
though the db name depends on what you have configured in `local.yaml`.
Safer to just run the database creation command in the `m1_mac_sqlite` directory, then also you don't overwrite your local dev database

mark-prins reviewed (approved):
👍  Wooo! it works 😁 

I think we go ahead and merge it in- it's useful to have this available

andreievg commented:
@mark-prins, this has been sitting for too long, I only just now had another look at it, updated with your change requests and moved commands into a script. Also, just added instructions of how to share with data vs using script.

andreievg commented on build/mac/README.md:
```diff
@@ -0,0 +1,29 @@
+
+Above will build and `bundle` files in `{..}_mac_sqlite` folder where `{..}` is replaced with architecture of the build
+
+In finder, right click on `{..}_mac_sqlite/open_msupply_server.sh`, then
```
As requested better instruction of `open with` modification

andreievg commented on build/mac/build.sh:
```diff
@@ -0,0 +1,16 @@
+#!/bin/bash
+
+# On M1 mac below would be DESTINATION="arm64_mac_sqlite"
+DESTINATION="$(uname -m)_mac_sqlite"
```
as requested use `uname` vs hard coded `m1_`

mark-prins commented:
yeah, I agree @andreievg time to merge it in. the changes look good thanks
